### PR TITLE
Fix :has_question_and_dashboard

### LINKED ISSUE #29438: :has_question_and_dashboard from `api/user/current` is always false
This query
https://github.com/metabase/metabase/blob/67bfd4dd62838ee45e895a42948ca78186c4ae06/src/metabase/api/user.clj#L207
is always returns false because `perms-query` is a map and `(perms-query user) => nil)`

### PR DESCRIPTION
**Pull request by qwef**
Closes https://github.com/metabase/metabase/issues/29438

### CODE REVIEW

File: test/metabase/api/user_test.clj [hunk 1]
```diff
@@ -303,7 +303,7 @@
                    (dissoc :is_qbnewb :last_login))
                (-> (mt/user-http-request :rasta :get 200 "user/current")
                    mt/boolean-ids-and-timestamps
-                   (dissoc :is_qbnewb :last_login :has_question_and_dashboard))))))
+                   (dissoc :is_qbnewb :has_question_and_dashboard :last_login))))))
     (testing "check that `has_question_and_dashboard` is `true`."
```
Comment (qwef) [inline test/metabase/api/user_test.clj:307]: I think this test should still test the new behavior

File: src/metabase/api/user.clj [hunk 2]
```diff
@@ -251,13 +250,10 @@
...
                           :collection_id
-                          (collection/permissions-set->visible-collection-ids @api/*current-user-permissions-set*))
-        perms-query {:where [:and
-                             [:= :archived false]
-                             coll-ids-filter]}]
-    #_{:clj-kondo/ignore [:discouraged-var]}
-    (assoc user :has_question_and_dashboard (and (db/exists? 'Card (perms-query user))
-                                                 (db/exists? 'Dashboard (perms-query user))))))
+                          (collection/permissions-set->visible-collection-ids @api/*current-user-permissions-set*))]
+    (assoc user :has_question_and_dashboard (and coll-ids-filter
```
Comment (noahmoss) [inline src/metabase/api/user.clj:254]: `coll-ids-filter` is just honeySQL that needs to be included in the `Card` and `Dashboard` queries; it's not a query in and of itself

File: test/metabase/api/user_test.clj (hunk 1 above)
Comment (noahmoss) [reply test/metabase/api/user_test.clj:307]: Do you know how it was working before? I thought that the issue was that it always returns `false`

**Review by noahmoss — CHANGES_REQUESTED** [review]
I think a simpler fix could just to change `(perms-query user)` to `perms-query` since it's not a function. But also fine with this refactor as long as `coll-ids-filter` still gets included in the query.

File: test/metabase/api/user_test.clj (hunk 1 above)
Comment (qwef) [reply test/metabase/api/user_test.clj:307]: Ah rip (db/exists? 'Card nil) returns true but (t2/exists? 'Card nil) returns false

**Review by noahmoss — APPROVED** [review]

**Comment by paoliniluis** [comment]
@qwef backport?

**Pull request merged**